src/controllers/QueryController.ts

```diff
@@ -168,10 +169,11 @@ export class QueryController extends RootComponent {
    * @returns {Promise<IQueryResults>}
    */
   public executeQuery(options?: IQueryOptions): Promise<IQueryResults> {
+    this.closeModalBox = options && options.closeModalBox != null ? options.closeModalBox : this.closeModalBox;
```

Comment:
I'd move this in a dedicated method, so that the code becomes something like : 

```
this.closeModalBoxIfNeeded(queryOptions)
```

or 

```
if (this.shouldCloseModalBox()) {
 this.modalBox.close()
}
```

Easier to read.